packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -250,6 +286,7 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
 
   debug(`Attempting to create ChangeSet with name ${changeSetName} to ${update ? 'update' : 'create'} stack ${deployName}`);
   print('%s: creating CloudFormation changeset...', colors.bold(deployName));
+  const executionId = uuid.v4();
   const changeSet = await cfn.createChangeSet({
     StackName: deployName,
     ChangeSetName: changeSetName,
```

Comment:
The defaulting of `rollback` should be done here. I can't comment on the line, because it wasn't changed, but it should be done on line 332.